content/docs/fundamentals/hmr.md

```diff
@@ -0,0 +1,126 @@
+# Hot Module Reloading
+
+Hot Module Reloading (HMR) is a feature baked in since AdonisJS 6.x.x that allows automatic reloading of modified files without having to restart the entire server.
+
+Until recently, the traditional way of working with AdonisJS was to launch the dev server with the command `node ace serve --watch`, which would take care of completely restarting the server/node process each time a file was modified, just like Nodemon does for example.
+
+This process works well, but depending on certain things, it's sometimes not the most optimal:
+
+- Sometimes you have a lot of configured packages, and therefore providers that execute on each restart. This can slow down the startup process.
+- If you use TSX as a template engine, then a simple modification of, let's say, paddings or margins in one of your views forces you to reload the entire server, resulting in a quite slow feedback loop.
+
+To correct these issues and improve the development experience, AdonisJS 6.x.x introduced Hot Module Reloading (HMR) based on [Hot-Hook](https://github.com/julien-R44/hot-hook), also developed by the Core Team.
+
+:::tip
+
+Make sure to read the extensive documentation of [Hot-Hook](https://github.com/julien-R44/hot-hook) to understand in more detail how it works.
+
+:::
+
+## Configuration
+
+First, make sure you have AdonisJS version 6.x.x. Then, you can configure HMR in your `package.json`. You will generally only need to specify the list of your controllers via a glob pattern.
+
+```jsonc
+// title: package.json
+{
+  "name": "adonis-app",
+  "type": "module",
+  // ...
+  "hotHook": {
+    "boundaries": [
+      "./app/controllers/**/*.ts"
+    ]
+  }
+}
+```
+
+:::tip
+What is a boundary? Make sure to read the [Hot-Hook documentation](https://github.com/Julien-R44/hot-hook#boundary) about this concept. 
+
+TL;DR: In Hot Hook, modules marked as "boundaries" create an area where HMR is applied, allowing for dynamic reloading of these modules and their dependencies as long as the dependency path to the entrypoint file includes at least one boundary-marked module. 
```

Comment:
> at least one boundary-marked module. 

That's not true; all paths to the root should go through a boundary-marked module to allow hot reloading. Otherwise, we are doing a cold reload.

Reply:
we are not talking about multiple paths here. Just one : 

> as long as the dependency path to the entrypoint file

i deliberately omitted the explanation when there are several paths to root. otherwise it's too complex. here it's a just quick TLDR